spec/aws-record/record/client_configuration_spec.rb

```diff
@@ -0,0 +1,52 @@
+# Copyright 2015-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+require 'spec_helper'
+
+module Aws
+  module Record
+    describe 'ClientConfiguration' do
+
+      context 'inheritance support for dynamodb client' do
+        let(:parent_model) do
+          Class.new do
+            include(Aws::Record)
+            integer_attr(:id, hash_key: true)
+          end
+        end
+
+        let(:child_model) do
+          Class.new(parent_model) do
+            include(Aws::Record)
+            string_attr(:foo)
```

Comment:
foo isn't really used, we can probably skip

Reply:
Done! Since we are only testing clients, that makes sense.